src/mococrw/p11_lib.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2022 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+extern "C" {
+#include <libp11.h>
+}
+
+namespace mococrw
+{
+namespace p11
+{
+namespace lib
+{
+/**
+ * Thin wrapper around the "naked" P11 library (namely libp11).
+ *
+ * Prefix all method names with 'P11_' to distinguish them from their
+ * LibP11 counterparts.
+ */
+class LibP11
+{
+public:
+    /* Initialisation */
+    static PKCS11_CTX *P11_PKCS11_CTX_new(void) noexcept;
+    static int P11_PKCS11_CTX_load(PKCS11_CTX *ctx, const char *ident) noexcept;
```

Comment:
Ah, maybe it was not clear - the `p11_lib.h` is a very raw and thin layer to LibP11. It takes a similar approach to what has been done for OpenSSL with the introduction of `openssl_lib.h`. In turn, `p11_wrap.h` uses `p11_lib.h` to enable C++ features. It is at this layers, where translation to shared pointer of  PKCS11_CTX takes place, along with the specification of a custom deleter. In future PRs, functions of `p11_wrap.h` will be used internally by MoCOCrW to implement interfaces like the one you mentioned (i.e., `*fromHSM`). If you are interested in a complete picture, see the RFC PR to get an idea https://github.com/bmwcarit/MoCOCrW/pull/106.

I updated the commit message to make it a bit clearer.

W.R.T why I did not add the user interface in this PR - I think this PR is already big and complex enough. It will be harder to review and get merged if more code is added, imo but I am happy to explain further if there are any questions.

Reply:
If _lib.h and _wrap.h are not supposed to be used by end-users, I wonder why do we not put those in src/mococrw/private folder then.

Reply:
I followed the directory path structure as openssl_lib.h, since they are similar